common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -156,6 +168,18 @@ private object Emitter {
 
 case class LogMarkerToken(component: String, action: String, state: String) {
     override def toString() = component + "_" + action + "_" + state
+
+    def asFinish = copy(state = LoggingMarkers.finish)
+    def asError = copy(state = LoggingMarkers.error)
+}
+
+object LogMarkerToken {
+    def parse(s: String) = {
+        // Per convention the components are guaranteed to not contain '_'
```

Comment:
for posterity; one option discussed on slack with @markusthoemmes is to lift two of the necessary `require` statements out to constructors for `component` and `state` values (by making those types other than String). Since these two are at least a closed set, one can enforce the invariant at construction time when the invoker is started.